jms/src/main/scala/akka/stream/alpakka/jms/impl/JmsConnector.scala

```diff
@@ -214,6 +215,12 @@ private[jms] trait JmsConnector[S <: JmsSession] {
   }
 
   override def onTimer(timerKey: Any): Unit = timerKey match {
+    case FlushAcknowledgementsTimerKey(session, timeout) =>
```

Comment:
This timer should be cancelled when the stage shuts down (or a session shuts down?)